source/deployment/ha.rst

```diff
@@ -0,0 +1,55 @@
+
+- Support for high availability database configurations is available in Mattermost Enterprise Edition E20 version 3.1 and higher. 
+- Support for high availability configurations of the Mattermost application server is planned for release in Mattermost Enterprise Edition E20 during summer 2016. 
+
+Setting up an active-passive redundant database configuration
+====
+
+The Mattermost server can set up to use one "master" database and up to 8 read replica databases. Mattermost distributes read requests across all databases, and sends write requests to the master database, and those changes are then sent to update the read replicas. 
+
+Sizing databases
+^^^^
+
+Please see `documentation on sizing database servers <http://docs.mattermost.com/install/requirements.html#hardware-requirements>`_ for guidance to determine appropriate hardware. 
+
+It's highly recommended that at least one read replica have a minimum of **three times the disk space** of the master database. This way, if the master database signals a "low disk space" warning the administrator can switch over to the read replica with a larger disk and continue operation, while also having room to locally duplicate and manipulate the production database. 
+
+
+Deploying a multi-database configuration 
+^^^^
+
+To configure a multi-database Mattermost server edit the `DataSource` setting `config.json` with a list of comma-separated database connection strings, with the first entry being the connection to the master database.
+
+All connection strings must be compatible with the database type set in `DriverName`, which would be either `postgres` or `mysql`. 
+
+Database write requests will be sent to the master database and read requests will be distributed among the other databases in the list. 
+Loading a multi-database configuration onto an active server
+^^^^
+
+After a multi-database configuration has been defined in `config.json` the following procedure can be used to apply the settings without shutting down the Mattermost server: 
+
+1. Go to **System Console** > **Configuration** and press *Reload Configuration from Disk** to reload configuration settings for the Mattermost server from `config.json`. 
+2. Go to **System Console** > **Database** and press **Recycle Database Connections** to takedown existing database connections and set up new connections in the multi-database configuration. 
+
+While connection settings are changing there will be a brief moment when writes to the master database will be unsuccessful. End users attempting to send messages while the switch is happening will have an experience similar to losing connection to the Mattermost server.
+
+Automatic read replica failover 
+^^^^
+
+Should a read replica fail from exceed disk space, having a hardware failure, a network disconnection, or any other reason, the Mattermost server will automatically re-distribute read requests to the other read replica and master database servers. 
```

Comment:
This is only true if you setup something like a mysql cluster and/or reload/recycle the config file.